app/views/posts/index.html.erb

```diff
@@ -1,7 +1,11 @@
+<% if current_user %>
 <% @posts.each do |post| %>
   <p><%= post.message %></p>
 <% end %>
 
 <%= link_to new_post_path do %>
   New post
 <% end %>
+<% else %>
+You must be signed in to view these posts
+<% end %>
```

Comment:
Do we still need these bits as it's covered by the before_action in the controller?